docs/guides/intro-to-graphql.md

```diff
@@ -1,9 +1,407 @@
+- `{Field}_starts_with`: String
+- `{Field}_not_starts_with`: String
+- `{Field}_ends_with`: String
+- `{Field}_not_ends_with`: String
+- `{Field}_i`: String
+- `{Field}_not_i`: String
+- `{Field}_contains_i`: String
+- `{Field}_not_contains_i`: String
+- `{Field}_starts_with_i`: String
+- `{Field}_not_starts_with_i`: String
+- `{Field}_ends_with_i`: String
+- `{Field}_not_ends_with_i`: String
+- `{Field}_in`: [String]
+- `{Field}_not_in`: [String]
+
+##### ID `where` filters
+
+- `{Field}`: ID
+- `{Field}_not`: ID
+- `{Field}_in`: [ID!]
+- `{Field}_not_in`: [ID!]
+
+##### Integer `where` filters
+
+- `{Field}: Int`
+- `{Field}_not`: Int
+- `{Field}_lt`: Int
+- `{Field}_lte`: Int
+- `{Field}_gt`: Int
+- `{Field}_gte`: Int
+- `{Field}_in`: [Int]
+- `{Field}_not_in`: [Int]
+
+#### Operators
+
+You can combine multiple where clauses with `AND` or `OR` operators.
+
+- `AND`: [whereInput]
+- `OR`: [whereInput]
+
+#### `search`
+
+Will search the list to limit results.
+
+```gql
+query {
+  allUsers(search: "Mike") {
+    id
+  }
+}
+```
+
+#### `orderBy`
+
+Order results. The orderBy string should match a field name in the List.
+
+```gql
+query {
+  allUsers(orderBy: "name") {
+    id
+  }
+}
+```
+
+#### `first`
+
+Limits the number of items returned from the query. Limits will be applied after `skip`, `orderBy`, `where` and `search` values are applied.
+
+If less results are available, the number of available results will be returned.
+
+```gql
+query {
+  allUsers(first: 10) {
+    id
+  }
+}
+```
+
+#### `skip`
+
+```gql
+query {
+  allUsers(skip: 10) {
+    id
+  }
+}
+```
+
+Skip the number of results specified. Is applied before `first` parameter, but after `orderBy`, `where` and `search` values.
+
+If the value of `skip` is greater than the number of available results, zero results will be returned.
+
+## Custom queries and Mutations
+
+You can add to the keystones generated schema with custom types, queries, and mutations using the `keystone.extendGraphQLSchema` method.
+
+### Usage
+
+```javascript
+keystone.extendGraphQLSchema({
+  types: ['type FooBar { foo: Int, bar: Float }'],
+  queries: [
+    {
+      schema: 'double(x: Int): Int',
+      resolver: (_, { x }) => 2 * x,
+    },
+  ],
+  mutations: [
+    {
+      schema: 'double(x: Int): Int',
+      resolver: (_, { x }) => 2 * x,
+    },
+  ],
+});
+```
+
+### Combining query arguments for pagination
+
+When `first` and `skip` are used together with the `count` from `_allUsersMeta`, this is sufficient to implement pagination on the list.
+
+It is important to provide the same `where` and `search` arguments to both the `allUser` and `_allUserMeta` queries. For example:
+
+```gql
+query {
+  allUsers (search:'a', skip: 10, first: 10) {
+    id
+  }
+  _allUsersMeta(search: 'a') {
+    count
+  }
+}
+```
+
+When `first` and `skip` are used together, skip works as an offset for the `first` argument. For example`(skip:10, first:10)` selects results 11 through 20.
```

Comment:
I like to avoid ambiguities in examples like this, so would suggest using different values for the different parameters:

```suggestion
When `first` and `skip` are used together, skip works as an offset for the `first` argument. For example`(skip:10, first:5)` selects results 11 through 25.
```

Reply:
The suggested change is wrong right? It would select `11` through `15`? I don't mind what numbers are used as and example, whatever is correct is fine.

Reply:
Should we also mention the effect of  `queryLimits` list config when searching, if any?